### PR TITLE
 ci: troubleshooting

### PR DESCRIPTION
**Pull request by 0x501D**
* ci: bump gevent version
* ci: drop Impish and Hirsute
* ci: drop macOS 10.15 from CI

### CODE REVIEW

File: .github/workflows/osx_testing.yml [hunk 1]
```diff
@@ -18,7 +18,6 @@ jobs:
       fail-fast: false
       matrix:
         runs-on:
-          - macos-10.15
```
Comment (Totktonada) [inline .github/workflows/osx_testing.yml:21]: You can add `macos-12` (beware: `macos-12`, not `macos-12.0`).

Comment (0x501D) [reply .github/workflows/osx_testing.yml:21]: Done

File: .github/workflows/osx_testing.yml [hunk 2]
```diff
@@ -178,7 +178,7 @@ jobs:
       - name: Install testing dependencies
         run: |
           cd "${SRCDIR}"
-          pip3 install -r test-run/requirements.txt
+          pip3 install PyYAML gevent==22.10.2
```
Comment (Totktonada) [inline .github/workflows/osx_testing.yml:181]: Shouldn't we bump it in test-run instead/as well?

**Review by Totktonada — APPROVED** [review]

File: .github/workflows/osx_testing.yml (hunk 2 above)
Comment (0x501D) [reply .github/workflows/osx_testing.yml:181]: gevent-21.* is not compatible with python-3.11 (installed on macos).
I'll double-check the relevance of these edits.

Comment (Totktonada) [reply .github/workflows/osx_testing.yml]: Yep (you said this in the commit message). But whether gevent-22.* incompatible with python <= 2.10?

Comment (0x501D) [reply .github/workflows/osx_testing.yml:181]: From gevent changelog:
'Support for legacy versions of Python, including 2.7 and 3.6, will be ending soon'

Comment (Totktonada) [reply .github/workflows/osx_testing.yml:181]: Okay, the reason is compatibility with python < 2.11.

It is interesting how it works for Mac OS 11/12 for other projects, at least Tarantool itself. But it is not a question for this patch, so I'm closing the discussion.

**Review by alyapunov — APPROVED** [review]

**Pull request merged**